crates/net/network/src/peers/manager.rs

```diff
@@ -327,14 +327,17 @@ impl PeersManager {
 
     /// Apply the corresponding reputation change to the given peer
     pub(crate) fn apply_reputation_change(&mut self, peer_id: &PeerId, rep: ReputationChangeKind) {
+        // if the reputation change is a reset, reset the reputation and return
+        if rep.is_reset() {
+            if let Some(peer) = self.peers.get_mut(peer_id) {
+                peer.reset_reputation();
+            }
+            return
```

Comment:
pedantic nit:
we only need to do `if let Some(peer) = self.peers.get_mut(peer_id) {` and can do the if is_reset check in that if branch?

Reply:
Oh yeah mb!